pkg/containertools/imagereader.go

```diff
@@ -150,52 +153,28 @@ func extractBundleManifests(layerTarball, outputDir string, tarReader *tar.Reade
 			return err
 		}
 
-		if header.Typeflag == tar.TypeReg {
-			if header.Name == layerTarball {
-				// Found the embedded tarball for the layer
-				layerReader := tar.NewReader(tarReader)
-
-				err = extractTarballToDir(outputDir, layerReader)
-				if err != nil {
-					return err
-				}
-
-				return nil
+		if header.Typeflag == tar.TypeReg && header.Name == layerTarball {
+			// Found the embedded tarball for the layer
+			decompressed, err := compression.DecompressStream(tarReader)
+			if err != nil {
+				return err
 			}
-		}
-	}
-}
+			_, err = archive.Apply(context.TODO(), outputDir, decompressed, archive.WithFilter(adjustPerms))
```

Comment:
This is really just using containerd to unpack all images now. This is probably fine, but it is different behavior then creating a container + exporting the fs with each separate tool. (for example, the actual layered fs can preserve the original permissions entirely, while we have to make everything owner-writable to unpack)